### PR TITLE
No more duplicate questions

### PR DESCRIPTION
**Pull request by spioupiou**
I completely forgot this afternoon that we already have a model which contains both the game_content_id and the instance_id: that's the round!!
So I just added a validation to the round model, the combination of game_content_id and instance_id must be unique.
Then in the controller I sample the game_contents until it finds a unique question that hasn't been used in the instance yet (highly inspired by @carlnoval do-break loop for passcode, thank you for the inspiration!)

Please test it out! (And please be careful because right now we only have 17 seeds, so if you set more than 17 rounds for your game, rails will try to find a unique question and loop indefinitely. Tested by yours truly 🤣 )

Screen Shot 2022-02-26 at 22 06 08

### CODE REVIEW

**Comment by carlnoval** [comment]
> the combination of game_content_id and instance_id must be unique
smart 👍🏼 

> so if you set more than 17 rounds for your game, rails will try to find a unique question and loop indefinitely
right lol, I don't think we have to worry about this if we limit the rounds, if I recall correctly we'll have max of 5 rounds per game with impostor feature

File: app/controllers/rounds_controller.rb [hunk 1]
```diff
@@ -48,7 +48,22 @@ def show
...
+
+  private
+
+  def generate_unique_question(instance, round)
+    game = Game.find(instance.game_id)
+    game_contents_array = game.game_contents
+
+    loop do
+      game_contents_array = Game.game_contents if game_contents_array.empty?
+      round.game_content_id = game_contents_array.sample.id
```
Comment (carlnoval) [inline app/controllers/rounds_controller.rb:63]: Currently the `game_contents_array` will never be empty cause we're not `.pop`ing an instance out of it.

If you do `game_contents_array.shuffle!.pop.id`:
- `shuffle!` should re-order the elements inside the array
- `pop` should get the random element that ended up at being the last element of the array after the shuffle
- combination of the two creates a method like `sample!` which currently don't exist

Comment (carlnoval) [reply app/controllers/rounds_controller.rb:63]: Thanks @spioupiou, one last thing 🙏🏼 

Currently the `game_contents_array` will never be empty cause we're not `.pop`ing an instance out of it.

If you do `game_contents_array.shuffle!.pop.id`:
- `shuffle!` should re-order the elements inside the array
- `pop` should get the random element that ended up at being the last element of the array after the shuffle
- combination of the two creates a method like `sample!` which currently don't exist

**Comment by spioupiou** [comment]
Oh my you're absolutely right!! This is corrected now :)

**Comment by carlnoval** [comment]
Thanks for this fix too, merging!

**Pull request merged**